scripts/clang-format.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# pre-requisite: Docker
+
+[[ -z "$1" ]] && adb_path="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )" || adb_path=$(echo "$1" | sed 's:/*$::') # Get root directory of repo (can be passed as argument)
+[[ -z "$2" ]] && ent_dir="enterprise" || ent_dir=$(echo "$2" | sed 's:/*$::') # Get enterprise directory (can be passed as argument)
+
+echo "ArangoDB directory: $adb_path"
+echo "Enterprise directory: $adb_path/$ent_dir"
+
+cd "$adb_path"
+community_diff=$(git diff HEAD --diff-filter=ACMR --name-only -- '*.cpp' '*.hpp' '*.cc' '*.c' '*.h')
+
+if [ -d "$adb_path/$ent_dir" ] # assume enterprise directory is within arangodb directory
+then
+   cd "$adb_path/$ent_dir"
+   enterprise_diff=$(git diff HEAD --diff-filter=ACMR --name-only -- '*.cpp' '*.hpp' '*.cc' '*.c' '*.h' | sed "s,^,$ent_dir/,")
+fi
+
+diff="$community_diff $enterprise_diff"
+if ! [[ -z "${diff// }" ]]; then
+  docker run --rm -v "$adb_path":/usr/src/arangodb arangodb/clang-format:1.0 "$diff"
```

Comment:
On Linux systems, this will change ownership of all files touched to `root.root`. This would be a nuisance. However, it is easy to provide a second script which runs `clang-format` locally without docker, solving the problem.

Reply:
Running the container with `--user $(id -u):$(id -g)` should avoid this ownership problem.

Reply:
I've created a separate [clang-format-linux.sh](https://github.com/arangodb/arangodb/pull/15083/files#diff-fe2859c9145b46a037443925b6569a707229f36e1353d4e227eb06513efa50c6) file to include the changes we discussed.

This script actually works fine on Mac as well (tested it locally). If it happens to work on Windows too, then there will be no need to have 2 different shell scripts

Reply:
Yes, I would think so. However, changes are probably not great, but I am no Windows expert.